content/docs/guides/neon-authorize.md

```diff
@@ -0,0 +1,150 @@
+
+![neon authorize architecture](/docs/guides/neon_authorize_architecture.png)
+
+## Before and after Neon Authorize
+
+Let's take a **before/after** look at moving authorization from the application level to the database to demonstrate how Neon Authorize offers a different approach to securing your application.
+
+### Before Neon Authorize (application-level checks):
+
+In a traditional setup, you might handle authorization for a function directly in your backend code:
+
+```typescript shouldWrap
+export async function insertTodo(newTodo: { newTodo: string; userId: string }) {
+  const { userId } = auth(); // Gets the user's ID from the JWT or session
+
+  if (!userId) throw new Error('No user logged in'); // No user authenticated
+
+  if (newTodo.userId !== userId) throw new Error('Unauthorized'); // User mismatch
+
+  // Inserts the new todo, linking it to the authenticated user
+  await fetchWithDrizzle(async (db) => {
+    return db.insert(schema.todos).values({
+      task: newTodo.newTodo,
+      isComplete: false,
+      userId, // Explicitly ties todo to the user
+    });
+  });
+
+  revalidatePath('/');
+}
+```
+
+In this case, you have to:
+
+- Check if the user is authenticated and their `userId` matches the data they are trying to modify.
+- Handle both task creation and authorization in the backend code.
+
+### After Neon Authorize (RLS in the database):
+
+With Neon Authorize, you can let the database handle the authorization through **Row-Level Security** (RLS) policies:
+
+```typescript
+pgPolicy('create todos', {
+  for: 'insert',
+  to: 'authenticated',
+  withCheck: sql`(select auth.user_id() = user_id)`,
+});
+```
+
+Now, in your backend, you can simplify the logic, removing the user authentication checks and explicit authorization handling.
+
+```typescript shouldWrap
+export async function insertTodo(newTodo: { newTodo: string }) {
+  await fetchWithDrizzle(async (db) => {
+    return db.insert(schema.todos).values({
+      task: newTodo.newTodo,
+      isComplete: false,
+    });
+  });
+
+  revalidatePath('/');
+}
+```
+
+This approach is flexible: you can manage RLS policies directly in SQL or work more declaratively using an ORM, applying authorization within your schema. This ability to handle both schema and authorization in one place can make it easier to maintain security.
```

Comment:
Is it only Drizzle that will support declarative RLS at this point? If so, should we say " ...and ORM that supports RLS..."
And maybe this is for a separate guide, but we should link to the drizzle example somewhere when it's ready.
https://neondb.slack.com/archives/C066CAC2NQK/p1729169058382619

Reply:
Good point, I'll reword. 

And the Drizzle RLS sample: yes, that looks like maybe a separate guide or docs page. We need a deeper dive into the authenticated/anonymous roles and how you'd use them.